### DISCUSSION TITLE
Debugging HTML::Mason

### DISCUSSION THREAD

**Discussion by shahramatrian** [Camelcade/Perl5-IDEA#2435] [Q&A]
Hi. I'm a Perl newbie so please take it easy if my question looks silly. 
Is there any way to debug HTML::Mason scripts? I couldn't set breakpoints on those scripts, so here is what I do:

1) I run the application from IDE in debug mode ([Best Practical RT](https://bestpractical.com/request-tracker) in this case)
2) I set breakpoints on corresponding compiled files (with .obj extension). For example /opt/rt5/var/mason_data/obj/3642352085/standard/**Ticket/Display.html**.obj
3) I open related URL page (<domain-url>/**Ticket/Display.html**<query-params>

I expect the debugger to stop on a breakpoint but it doesn't although breakpoints on other Perl scripts and packages work properly.
Can you please help?

**Comment by hurricup** [comment]
It doesn't work out of the box and requires support from the framework side, HTML::Mason in this case. See https://metacpan.org/pod/Devel::Camelcadedb#TEMPLATE-ENGINES-INTEGRATION

**Reply by hurricup** [reply]
Perl's breakpoints are just an elements of hash:
```
{
   '/foo/bar/baz.pl' => 5,
} 
```
And perl knows which code is from which file. This is how breakpoints works.
I presume that these obj files are loaded by mason (not perl itself) and evaled. So actually they are all some kind of `eval(123)` instead of real path. 
Ide sends breakpoints to debugger as is, but debugger may do the mapping.
